lib/os/fdtable.c

```diff
@@ -389,7 +389,14 @@ int zvfs_close(int fd)
 	(void)k_mutex_lock(&fdtable[fd].lock, K_FOREVER);
 	if (fdtable[fd].vtable->close != NULL) {
 		/* close() is optional - e.g. stdinout_fd_op_vtable */
-		res = fdtable[fd].vtable->close(fdtable[fd].obj);
+		if (fdtable[fd].mode & ZVFS_MODE_CLOSE2) {
```

Comment:
To me, this seems like all that should be necessary.

```suggestion
		if (fdtable[fd].mode & ZVFS_MODE_IFSOCK) {
```